src/test/bufferlist.cc

```diff
@@ -2284,7 +2284,7 @@ TEST(BufferList, append_zero) {
   EXPECT_EQ((unsigned)1, bl.get_num_buffers());
   EXPECT_EQ((unsigned)1, bl.length());
   bl.append_zero(1);
-  EXPECT_EQ((unsigned)2, bl.get_num_buffers());
+  EXPECT_EQ((unsigned)1, bl.get_num_buffers());
```

Comment:
Maybe merge this into the change to append_zero. Just a nit, but I like to have changes and relevant tests in the same commit where possible.

Not necessary, though.